src/main/java/chylex/hee/api/message/element/ListValue.java

```diff
@@ -31,17 +32,26 @@ public boolean checkType(NBTBase tag){
 
 	@Override
 	public boolean checkValue(NBTBase tag){
-		for(NBTBase element:(List<NBTBase>)((NBTTagList)tag).tagList){
-			if (!condition.checkType(element) || !condition.checkValue(element))return false;
-		}
-		
+		List tagList = ReflectionUtils.getFieldValue((List<NBTBase>)((NBTTagList)tag), "tagList");
```

Comment:
There is no reason to use reflection here either way. The interface for iterating over NBTTagList content is quite cumbersome, but it does exist. Might use that instead.